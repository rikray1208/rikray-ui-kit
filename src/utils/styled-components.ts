import * as styledComponents from 'styled-components';
import {ITheme} from "../theme/types"


const {
    default: styled,
    css,
    createGlobalStyle,
    keyframes,
    ThemeProvider,
} = styledComponents as styledComponents.ThemedStyledComponentsModule<ITheme>;

export { css, createGlobalStyle, keyframes, ThemeProvider };
export default styled;
