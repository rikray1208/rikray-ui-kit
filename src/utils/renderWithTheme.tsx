import {render} from "@testing-library/react";
import React from "react";
import {ITheme} from "../theme/types";
import {ThemeProvider} from "./styled-components";

export const renderWithTheme = (component: React.ReactNode, theme: ITheme) => {
    return render(
        <ThemeProvider theme={theme}>
            {component}
        </ThemeProvider>
    )
}
