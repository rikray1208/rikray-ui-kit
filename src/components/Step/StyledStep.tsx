import styled, { css } from 'styled-components';

import { State, StepProps } from './index';

import { setFontSize } from '../../utils/helpers';
export const StepWrapper = styled.div<{ vertical: boolean }>`
    display: flex;
    width: 100%;
    ${({ vertical }) => vertical && css`flex-direction: column; width: fit-content;'`}
`;
export const StepItem = styled.div<{ vertical: boolean }>`
    ${({ vertical }) =>
        vertical
            ? css`
                  width: fit-content;
                  position: relative;
              `
            : css`
                  display: flex;
                  align-items: center;
                  flex: 1;
                  gap: 10px;
              `}
`;
export const StepIcon = styled.div<
    Pick<StepProps, 'small' | 'vertical'> & { isIcon: boolean; state: number; vertical: boolean }
>`
    ${({ theme, isIcon, small, state, vertical }) => css`
        display: flex;
        justify-content: center;
        align-items: center;

        padding: ${theme.Step['padding-1.5'] + 'rem'}
            ${isIcon ? theme.Step['padding-1.5'] + 'rem' : theme.Step['padding-2.5'] + 'rem'};
        ${small &&
        css`
            padding: ${theme.Step['padding-1'] + 'rem'}
                ${isIcon ? theme.Step['padding-1'] + 'rem' : theme.Step['padding-2'] + 'rem'};
        `}

        ${setFontSize(small ? 'Medium' : 'Large', 'Step', {})}
    background: ${theme.Step.colorBgSecondary};
        background: ${(state == State.CURRENT || state == State.LESS) && theme.Step.colorPrimary};
        border-radius: 50%;
        transition: background ${theme.Step.durationFast + 'ms'} ${theme.Step.transitionEaseInOut};

        color: ${theme.Step.colorWhite};
        white-space: pre;

        ${vertical &&
        css`
            float: left;
            margin-right: 12px;
        `}
    `}
`;
export const StepContent = styled.div<{ isLast: boolean; vertical: boolean }>`
    ${({ theme, isLast, vertical }) => css`
        ${isLast &&
        css`
            min-width: ${theme.Step.descriptionWidth + 'px'};
        `};
        ${vertical
            ? css`
                  overflow: hidden;
              `
            : css`
                  position: relative;
              `}
    `}
`;
export const StepTitle = styled.div<Pick<StepProps, 'small'> & { state: number }>`
    ${({ theme, small, state }) => css`
        ${setFontSize('Large', 'Step', { greater: !small })};
        color: ${state == State.LESS || state == State.CURRENT
            ? theme.Step.colorTextBase
            : theme.Step.colorTextTertiary};
        white-space: nowrap;
        height: 100%;
    `}
`;

export const StepDescription = styled.div<Pick<StepProps, 'small'> & { state: number; vertical: boolean }>`
    ${({ theme, small, state, vertical }) => css`
        ${!vertical &&
        css`
            position: absolute;
        `}
        width: ${vertical ? theme.Step.descriptionVerticalWidth + 'px' : theme.Step.descriptionWidth + 'px'};

        ${setFontSize('Large', 'Step', { less: small })};
        color: ${state == State.LESS || state == State.CURRENT
            ? theme.Step.colorTextBase
            : theme.Step.colorTextTertiary};
        font-weight: ${theme.Step.fontWeightLight};
    `}
`;
export const StepLine = styled.div<{ state: number; vertical: boolean; small: boolean }>`
    ${({ theme, state, vertical, small }) => css`
        ${vertical
            ? css`
                  position: absolute;
                  height: 100%;
                  top: 0;
                  left: 17px;
                  padding: 39px 0 3px;

                  ${small &&
                  css`
                      left: 14px;
                      padding: 32px 0 4px;
                  `}
              `
            : css`
                  position: relative;
                  margin-right: 10px;
                  width: 100%;
              `}

        &::after {
            content: '';
            ${vertical
                ? css`
                      display: block;
                      width: 1px;
                      height: 100%;
                      border-radius: 1px;
                  `
                : css`
                      position: absolute;
                      height: 1px;
                      width: 100%;
                  `}
            border-radius: 1px;
            transition: all ${theme.Step.durationSlow + 'ms'} ${theme.Step.transitionEaseInOut};
            background: ${theme.Step.colorBgSecondary};
            background: ${(state == State.CURRENT || state == State.LESS) && theme.Step.colorPrimary};
        }
    `}
`;
