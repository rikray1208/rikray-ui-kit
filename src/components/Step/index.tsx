import { Check } from 'lucide-react';
import React, { FC } from 'react';

import { StepContent, StepDescription, StepIcon, StepItem, StepLine, StepTitle, StepWrapper } from './StyledStep';
export interface StepElement {
    title: string;
    description?: string;
    icon?: React.ReactNode;
}
export interface StepProps {
    current: number;
    steps: StepElement[];
    vertical?: boolean;
    small?: boolean;
}
export enum State {
    LESS = -1,
    CURRENT = 0,
    GREATER = 1,
}
const Step: FC<StepProps> = (props) => {
    const { current, vertical = false, steps, small = false } = props;
    function validatedNumber(index: number): string {
        return index < 10 ? ' ' + index + ' ' : index + '';
    }
    function stepState(index: number): number {
        if (index < current) return State.LESS;
        if (index > current) return State.GREATER;
        return State.CURRENT;
    }
    return (
        <>
            <StepWrapper vertical={vertical}>
                {steps.map((step, index) => (
                    <StepItem
                        id={'item'}
                        key={step.title}
                        style={index == steps.length - 1 ? { flex: '0 1 0' } : {}}
                        vertical={vertical}
                    >
                        <StepIcon
                            id={'step number'}
                            isIcon={index < current}
                            small={small}
                            state={stepState(index)}
                            vertical={vertical}
                        >
                            {index < current ? <Check size={small ? 21 : 24} /> : validatedNumber(index + 1)}
                        </StepIcon>
                        <StepContent id={'content'} isLast={index == steps.length - 1} vertical={vertical}>
                            <StepTitle id={'title'} small={small} state={stepState(index)}>
                                {step.title}
                            </StepTitle>
                            <StepDescription
                                id={'description'}
                                small={small}
                                state={stepState(index)}
                                vertical={vertical}
                            >
                                {step.description}
                            </StepDescription>
                        </StepContent>

                        {index !== steps.length - 1 && (
                            <StepLine id={'step line'} small={small} vertical={vertical} state={stepState(index + 1)} />
                        )}
                    </StepItem>
                ))}
            </StepWrapper>
        </>
    );
};

export default Step;
