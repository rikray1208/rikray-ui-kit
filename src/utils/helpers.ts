import {Shape, Size} from "../components/types";
import {css} from "styled-components";
import React from "react";
import renderer from "react-test-renderer";
import {ComponentMap} from "../theme/types";

type Component = keyof  ComponentMap;

export function setShape(shape: Shape, Component: Component) {
    if (shape === 'Round')
        return css`border-radius: ${({theme}) => theme[Component]["borderRadius-lg"] + 'px'};`

    if (shape === 'SemiRound')
        return css`border-radius: ${({theme}) => theme[Component]["borderRadius-sm"] + 'px'};`

    if (shape === 'Rectangle')
        return css`border-radius: ${({theme}) => theme[Component]["borderRadius-xs"] + 'px'};`
}
export function setFontSize(size: Size, Component: Component) {
    if (size == 'Small')
        return css`
          font-size: ${({theme}) => theme[Component]["fontSize-sm"] + 'rem'};
          line-height: ${({theme}) => theme[Component]["lineHeight-sm"] + 'rem'};
        `

    if (size == 'Medium')
        return css`
          font-size: ${({theme}) => theme[Component]["fontSize-base"] + 'rem'};
          line-height: ${({theme}) => theme[Component]["lineHeight-base"] + 'rem'};
        `

    if (size == 'Large')
        return css`
          font-size: ${({theme}) => theme[Component]["fontSize-lg"] + 'rem'};
          line-height: ${({theme}) => theme[Component]["lineHeight-lg"] + 'rem'};
        `
}

export const assertReactElement = (element: React.ReactElement) => {
    const component = renderer.create(element);
    return component.toJSON();
};
