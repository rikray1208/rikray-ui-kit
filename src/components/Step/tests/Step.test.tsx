import React from "react";
import 'jest-styled-components'
import {ThemeProvider} from "../../../index";
import {assertReactElement} from "../../../utils/helpers";
import Step, {StepElement} from "../index";

const Steps: StepElement[] = [
    {title: 'Login', description:'This is a description'},
    {title: 'Password asd', description:'This is a description'},
    {title: 'Auth Code', description:'This is a description'},
]

describe('Step', () => {
    it('should return a Step', () =>{
        const element = (
            <ThemeProvider><Step current={0} steps={Steps}/></ThemeProvider>
        );
        expect(assertReactElement(element)).toMatchSnapshot();
    })
    it('should return a Step with one success element', () =>{
        const element = (
            <ThemeProvider><Step current={1} steps={Steps}/></ThemeProvider>
        );
        expect(assertReactElement(element)).toMatchSnapshot();
    })
    it('should return a small Step', () =>{
        const element = (
            <ThemeProvider><Step current={0} steps={Steps} small={true}/></ThemeProvider>
        );
        expect(assertReactElement(element)).toMatchSnapshot();
    })
    it('should return a vertical Step', () =>{
        const element = (
            <ThemeProvider><Step current={0} steps={Steps} vertical={true}/></ThemeProvider>
        );
        expect(assertReactElement(element)).toMatchSnapshot();
    })
    it('should return a small vertical Step', () =>{
        const element = (
            <ThemeProvider><Step current={0} steps={Steps} vertical={true} small={true}/></ThemeProvider>
        );
        expect(assertReactElement(element)).toMatchSnapshot();
    })
})


