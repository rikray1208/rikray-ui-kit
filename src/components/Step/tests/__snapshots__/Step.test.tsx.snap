// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Step should return a Step 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  gap: 10px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0.42500000000000004rem 0.675rem;
  font-size: 1.125rem;
  line-height: 1.75rem;
  background: #dedede;
  background: #3B74F3;
  border-radius: 50%;
  -webkit-transition: background 100ms cubic-bezier(0.4,0,0.2,1);
  transition: background 100ms cubic-bezier(0.4,0,0.2,1);
  color: #ffffff;
  white-space: pre;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0.42500000000000004rem 0.675rem;
  font-size: 1.125rem;
  line-height: 1.75rem;
  background: #dedede;
  border-radius: 50%;
  -webkit-transition: background 100ms cubic-bezier(0.4,0,0.2,1);
  transition: background 100ms cubic-bezier(0.4,0,0.2,1);
  color: #ffffff;
  white-space: pre;
}

.c3 {
  position: relative;
}

.c10 {
  min-width: 200px;
  position: relative;
}

.c4 {
  font-size: 1.25rem;
  line-height: 2rem;
  color: #000000;
  white-space: nowrap;
  height: 100%;
}

.c8 {
  font-size: 1.25rem;
  line-height: 2rem;
  color: rgba(0,0,0,0.50);
  white-space: nowrap;
  height: 100%;
}

.c5 {
  position: absolute;
  width: 200px;
  font-size: 1.125rem;
  line-height: 1.75rem;
  color: #000000;
  font-weight: 300;
}

.c9 {
  position: absolute;
  width: 200px;
  font-size: 1.125rem;
  line-height: 1.75rem;
  color: rgba(0,0,0,0.50);
  font-weight: 300;
}

.c6 {
  position: relative;
  margin-right: 10px;
  width: 100%;
}

.c6::after {
  content: "";
  position: absolute;
  height: 1px;
  width: 100%;
  border-radius: 1px;
  -webkit-transition: all 300ms cubic-bezier(0.4,0,0.2,1);
  transition: all 300ms cubic-bezier(0.4,0,0.2,1);
  background: #dedede;
}

<div
  className="c0"
>
  <div
    className="c1"
    id="item"
    style={{}}
  >
    <div
      className="c2"
      id="step number"
    >
       1 
    </div>
    <div
      className="c3"
      id="content"
    >
      <div
        className="c4"
        id="title"
      >
        Login
      </div>
      <div
        className="c5"
        id="description"
      >
        This is a description
      </div>
    </div>
    <div
      className="c6"
      id="step line"
    />
  </div>
  <div
    className="c1"
    id="item"
    style={{}}
  >
    <div
      className="c7"
      id="step number"
    >
       2 
    </div>
    <div
      className="c3"
      id="content"
    >
      <div
        className="c8"
        id="title"
      >
        Password asd
      </div>
      <div
        className="c9"
        id="description"
      >
        This is a description
      </div>
    </div>
    <div
      className="c6"
      id="step line"
    />
  </div>
  <div
    className="c1"
    id="item"
    style={
      {
        "flex": "0 1 0",
      }
    }
  >
    <div
      className="c7"
      id="step number"
    >
       3 
    </div>
    <div
      className="c10"
      id="content"
    >
      <div
        className="c8"
        id="title"
      >
        Auth Code
      </div>
      <div
        className="c9"
        id="description"
      >
        This is a description
      </div>
    </div>
  </div>
</div>
`;

exports[`Step should return a Step with one success element 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  gap: 10px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0.42500000000000004rem 0.42500000000000004rem;
  font-size: 1.125rem;
  line-height: 1.75rem;
  background: #dedede;
  background: #3B74F3;
  border-radius: 50%;
  -webkit-transition: background 100ms cubic-bezier(0.4,0,0.2,1);
  transition: background 100ms cubic-bezier(0.4,0,0.2,1);
  color: #ffffff;
  white-space: pre;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0.42500000000000004rem 0.675rem;
  font-size: 1.125rem;
  line-height: 1.75rem;
  background: #dedede;
  background: #3B74F3;
  border-radius: 50%;
  -webkit-transition: background 100ms cubic-bezier(0.4,0,0.2,1);
  transition: background 100ms cubic-bezier(0.4,0,0.2,1);
  color: #ffffff;
  white-space: pre;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0.42500000000000004rem 0.675rem;
  font-size: 1.125rem;
  line-height: 1.75rem;
  background: #dedede;
  border-radius: 50%;
  -webkit-transition: background 100ms cubic-bezier(0.4,0,0.2,1);
  transition: background 100ms cubic-bezier(0.4,0,0.2,1);
  color: #ffffff;
  white-space: pre;
}

.c3 {
  position: relative;
}

.c10 {
  min-width: 200px;
  position: relative;
}

.c4 {
  font-size: 1.25rem;
  line-height: 2rem;
  color: #000000;
  white-space: nowrap;
  height: 100%;
}

.c11 {
  font-size: 1.25rem;
  line-height: 2rem;
  color: rgba(0,0,0,0.50);
  white-space: nowrap;
  height: 100%;
}

.c5 {
  position: absolute;
  width: 200px;
  font-size: 1.125rem;
  line-height: 1.75rem;
  color: #000000;
  font-weight: 300;
}

.c12 {
  position: absolute;
  width: 200px;
  font-size: 1.125rem;
  line-height: 1.75rem;
  color: rgba(0,0,0,0.50);
  font-weight: 300;
}

.c6 {
  position: relative;
  margin-right: 10px;
  width: 100%;
}

.c6::after {
  content: "";
  position: absolute;
  height: 1px;
  width: 100%;
  border-radius: 1px;
  -webkit-transition: all 300ms cubic-bezier(0.4,0,0.2,1);
  transition: all 300ms cubic-bezier(0.4,0,0.2,1);
  background: #dedede;
  background: #3B74F3;
}

.c8 {
  position: relative;
  margin-right: 10px;
  width: 100%;
}

.c8::after {
  content: "";
  position: absolute;
  height: 1px;
  width: 100%;
  border-radius: 1px;
  -webkit-transition: all 300ms cubic-bezier(0.4,0,0.2,1);
  transition: all 300ms cubic-bezier(0.4,0,0.2,1);
  background: #dedede;
}

<div
  className="c0"
>
  <div
    className="c1"
    id="item"
    style={{}}
  >
    <div
      className="c2"
      id="step number"
    >
      <svg
        className="lucide lucide-check"
        fill="none"
        height={24}
        stroke="currentColor"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth={2}
        viewBox="0 0 24 24"
        width={24}
        xmlns="http://www.w3.org/2000/svg"
      >
        <polyline
          points="20 6 9 17 4 12"
        />
      </svg>
    </div>
    <div
      className="c3"
      id="content"
    >
      <div
        className="c4"
        id="title"
      >
        Login
      </div>
      <div
        className="c5"
        id="description"
      >
        This is a description
      </div>
    </div>
    <div
      className="c6"
      id="step line"
    />
  </div>
  <div
    className="c1"
    id="item"
    style={{}}
  >
    <div
      className="c7"
      id="step number"
    >
       2 
    </div>
    <div
      className="c3"
      id="content"
    >
      <div
        className="c4"
        id="title"
      >
        Password asd
      </div>
      <div
        className="c5"
        id="description"
      >
        This is a description
      </div>
    </div>
    <div
      className="c8"
      id="step line"
    />
  </div>
  <div
    className="c1"
    id="item"
    style={
      {
        "flex": "0 1 0",
      }
    }
  >
    <div
      className="c9"
      id="step number"
    >
       3 
    </div>
    <div
      className="c10"
      id="content"
    >
      <div
        className="c11"
        id="title"
      >
        Auth Code
      </div>
      <div
        className="c12"
        id="description"
      >
        This is a description
      </div>
    </div>
  </div>
</div>
`;

exports[`Step should return a small Step 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  gap: 10px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0.42500000000000004rem 0.675rem;
  padding: 0.30000000000000004rem 0.55rem;
  font-size: 1rem;
  line-height: 1.5rem;
  background: #dedede;
  background: #3B74F3;
  border-radius: 50%;
  -webkit-transition: background 100ms cubic-bezier(0.4,0,0.2,1);
  transition: background 100ms cubic-bezier(0.4,0,0.2,1);
  color: #ffffff;
  white-space: pre;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0.42500000000000004rem 0.675rem;
  padding: 0.30000000000000004rem 0.55rem;
  font-size: 1rem;
  line-height: 1.5rem;
  background: #dedede;
  border-radius: 50%;
  -webkit-transition: background 100ms cubic-bezier(0.4,0,0.2,1);
  transition: background 100ms cubic-bezier(0.4,0,0.2,1);
  color: #ffffff;
  white-space: pre;
}

.c3 {
  position: relative;
}

.c10 {
  min-width: 200px;
  position: relative;
}

.c4 {
  font-size: 1.125rem;
  line-height: 1.75rem;
  color: #000000;
  white-space: nowrap;
  height: 100%;
}

.c8 {
  font-size: 1.125rem;
  line-height: 1.75rem;
  color: rgba(0,0,0,0.50);
  white-space: nowrap;
  height: 100%;
}

.c5 {
  position: absolute;
  width: 200px;
  font-size: 1rem;
  line-height: 1.5rem;
  color: #000000;
  font-weight: 300;
}

.c9 {
  position: absolute;
  width: 200px;
  font-size: 1rem;
  line-height: 1.5rem;
  color: rgba(0,0,0,0.50);
  font-weight: 300;
}

.c6 {
  position: relative;
  margin-right: 10px;
  width: 100%;
}

.c6::after {
  content: "";
  position: absolute;
  height: 1px;
  width: 100%;
  border-radius: 1px;
  -webkit-transition: all 300ms cubic-bezier(0.4,0,0.2,1);
  transition: all 300ms cubic-bezier(0.4,0,0.2,1);
  background: #dedede;
}

<div
  className="c0"
>
  <div
    className="c1"
    id="item"
    style={{}}
  >
    <div
      className="c2"
      id="step number"
    >
       1 
    </div>
    <div
      className="c3"
      id="content"
    >
      <div
        className="c4"
        id="title"
      >
        Login
      </div>
      <div
        className="c5"
        id="description"
      >
        This is a description
      </div>
    </div>
    <div
      className="c6"
      id="step line"
    />
  </div>
  <div
    className="c1"
    id="item"
    style={{}}
  >
    <div
      className="c7"
      id="step number"
    >
       2 
    </div>
    <div
      className="c3"
      id="content"
    >
      <div
        className="c8"
        id="title"
      >
        Password asd
      </div>
      <div
        className="c9"
        id="description"
      >
        This is a description
      </div>
    </div>
    <div
      className="c6"
      id="step line"
    />
  </div>
  <div
    className="c1"
    id="item"
    style={
      {
        "flex": "0 1 0",
      }
    }
  >
    <div
      className="c7"
      id="step number"
    >
       3 
    </div>
    <div
      className="c10"
      id="content"
    >
      <div
        className="c8"
        id="title"
      >
        Auth Code
      </div>
      <div
        className="c9"
        id="description"
      >
        This is a description
      </div>
    </div>
  </div>
</div>
`;

exports[`Step should return a small vertical Step 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
}

.c1 {
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  position: relative;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0.42500000000000004rem 0.675rem;
  padding: 0.30000000000000004rem 0.55rem;
  font-size: 1rem;
  line-height: 1.5rem;
  background: #dedede;
  background: #3B74F3;
  border-radius: 50%;
  -webkit-transition: background 100ms cubic-bezier(0.4,0,0.2,1);
  transition: background 100ms cubic-bezier(0.4,0,0.2,1);
  color: #ffffff;
  white-space: pre;
  float: left;
  margin-right: 12px;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0.42500000000000004rem 0.675rem;
  padding: 0.30000000000000004rem 0.55rem;
  font-size: 1rem;
  line-height: 1.5rem;
  background: #dedede;
  border-radius: 50%;
  -webkit-transition: background 100ms cubic-bezier(0.4,0,0.2,1);
  transition: background 100ms cubic-bezier(0.4,0,0.2,1);
  color: #ffffff;
  white-space: pre;
  float: left;
  margin-right: 12px;
}

.c3 {
  overflow: hidden;
}

.c10 {
  min-width: 200px;
  overflow: hidden;
}

.c4 {
  font-size: 1.125rem;
  line-height: 1.75rem;
  color: #000000;
  white-space: nowrap;
  height: 100%;
}

.c8 {
  font-size: 1.125rem;
  line-height: 1.75rem;
  color: rgba(0,0,0,0.50);
  white-space: nowrap;
  height: 100%;
}

.c5 {
  width: 350px;
  font-size: 1rem;
  line-height: 1.5rem;
  color: #000000;
  font-weight: 300;
}

.c9 {
  width: 350px;
  font-size: 1rem;
  line-height: 1.5rem;
  color: rgba(0,0,0,0.50);
  font-weight: 300;
}

.c6 {
  position: absolute;
  height: 100%;
  top: 0;
  left: 17px;
  padding: 39px 0 3px;
  left: 14px;
  padding: 32px 0 4px;
}

.c6::after {
  content: "";
  display: block;
  width: 1px;
  height: 100%;
  border-radius: 1px;
  border-radius: 1px;
  -webkit-transition: all 300ms cubic-bezier(0.4,0,0.2,1);
  transition: all 300ms cubic-bezier(0.4,0,0.2,1);
  background: #dedede;
}

<div
  className="c0"
>
  <div
    className="c1"
    id="item"
    style={{}}
  >
    <div
      className="c2"
      id="step number"
    >
       1 
    </div>
    <div
      className="c3"
      id="content"
    >
      <div
        className="c4"
        id="title"
      >
        Login
      </div>
      <div
        className="c5"
        id="description"
      >
        This is a description
      </div>
    </div>
    <div
      className="c6"
      id="step line"
    />
  </div>
  <div
    className="c1"
    id="item"
    style={{}}
  >
    <div
      className="c7"
      id="step number"
    >
       2 
    </div>
    <div
      className="c3"
      id="content"
    >
      <div
        className="c8"
        id="title"
      >
        Password asd
      </div>
      <div
        className="c9"
        id="description"
      >
        This is a description
      </div>
    </div>
    <div
      className="c6"
      id="step line"
    />
  </div>
  <div
    className="c1"
    id="item"
    style={
      {
        "flex": "0 1 0",
      }
    }
  >
    <div
      className="c7"
      id="step number"
    >
       3 
    </div>
    <div
      className="c10"
      id="content"
    >
      <div
        className="c8"
        id="title"
      >
        Auth Code
      </div>
      <div
        className="c9"
        id="description"
      >
        This is a description
      </div>
    </div>
  </div>
</div>
`;

exports[`Step should return a vertical Step 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
}

.c1 {
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  position: relative;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0.42500000000000004rem 0.675rem;
  font-size: 1.125rem;
  line-height: 1.75rem;
  background: #dedede;
  background: #3B74F3;
  border-radius: 50%;
  -webkit-transition: background 100ms cubic-bezier(0.4,0,0.2,1);
  transition: background 100ms cubic-bezier(0.4,0,0.2,1);
  color: #ffffff;
  white-space: pre;
  float: left;
  margin-right: 12px;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0.42500000000000004rem 0.675rem;
  font-size: 1.125rem;
  line-height: 1.75rem;
  background: #dedede;
  border-radius: 50%;
  -webkit-transition: background 100ms cubic-bezier(0.4,0,0.2,1);
  transition: background 100ms cubic-bezier(0.4,0,0.2,1);
  color: #ffffff;
  white-space: pre;
  float: left;
  margin-right: 12px;
}

.c3 {
  overflow: hidden;
}

.c10 {
  min-width: 200px;
  overflow: hidden;
}

.c4 {
  font-size: 1.25rem;
  line-height: 2rem;
  color: #000000;
  white-space: nowrap;
  height: 100%;
}

.c8 {
  font-size: 1.25rem;
  line-height: 2rem;
  color: rgba(0,0,0,0.50);
  white-space: nowrap;
  height: 100%;
}

.c5 {
  width: 350px;
  font-size: 1.125rem;
  line-height: 1.75rem;
  color: #000000;
  font-weight: 300;
}

.c9 {
  width: 350px;
  font-size: 1.125rem;
  line-height: 1.75rem;
  color: rgba(0,0,0,0.50);
  font-weight: 300;
}

.c6 {
  position: absolute;
  height: 100%;
  top: 0;
  left: 17px;
  padding: 39px 0 3px;
}

.c6::after {
  content: "";
  display: block;
  width: 1px;
  height: 100%;
  border-radius: 1px;
  border-radius: 1px;
  -webkit-transition: all 300ms cubic-bezier(0.4,0,0.2,1);
  transition: all 300ms cubic-bezier(0.4,0,0.2,1);
  background: #dedede;
}

<div
  className="c0"
>
  <div
    className="c1"
    id="item"
    style={{}}
  >
    <div
      className="c2"
      id="step number"
    >
       1 
    </div>
    <div
      className="c3"
      id="content"
    >
      <div
        className="c4"
        id="title"
      >
        Login
      </div>
      <div
        className="c5"
        id="description"
      >
        This is a description
      </div>
    </div>
    <div
      className="c6"
      id="step line"
    />
  </div>
  <div
    className="c1"
    id="item"
    style={{}}
  >
    <div
      className="c7"
      id="step number"
    >
       2 
    </div>
    <div
      className="c3"
      id="content"
    >
      <div
        className="c8"
        id="title"
      >
        Password asd
      </div>
      <div
        className="c9"
        id="description"
      >
        This is a description
      </div>
    </div>
    <div
      className="c6"
      id="step line"
    />
  </div>
  <div
    className="c1"
    id="item"
    style={
      {
        "flex": "0 1 0",
      }
    }
  >
    <div
      className="c7"
      id="step number"
    >
       3 
    </div>
    <div
      className="c10"
      id="content"
    >
      <div
        className="c8"
        id="title"
      >
        Auth Code
      </div>
      <div
        className="c9"
        id="description"
      >
        This is a description
      </div>
    </div>
  </div>
</div>
`;
